src/download.rs

```diff
@@ -12,10 +15,53 @@ pub struct DownloadResult<W: std::io::Write> {
     pub data: W,
 }
 
+pub fn hash_on_disk_sha256(path: &Path, maxlen: Option<usize>) -> Result<omaha::Hash<omaha::Sha256>> {
+    let file = File::open(path).context(format!("failed to open path({:?})", path.display()))?;
+    let mut hasher = Sha256::new();
+
+    let filelen = file.metadata().unwrap().len() as usize;
```

Comment:
While at it, I think we should replace the unwrap by `.or_ok`

Reply:
I suppose you mean `.ok_or`, which requires `Option`.
In this case, `file.metadata()` returns not Option but `Error`.
So I would just use `.context` to specify errors.

Reply:
Yes, using `.context(E)?` would be good then.
For unwrap on Options we need https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or

Reply:
Done